sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSinkLog.scala

```diff
@@ -97,18 +97,13 @@ class FileStreamSinkLog(
     s"Please set ${SQLConf.FILE_SINK_LOG_COMPACT_INTERVAL.key} (was $defaultCompactInterval) " +
       "to a positive value.")
 
-  override def compactLogs(logs: Seq[SinkFileStatus]): Seq[SinkFileStatus] = {
-    val deletedFiles = logs.filter(_.action == FileStreamSinkLog.DELETE_ACTION).map(_.path).toSet
-    if (deletedFiles.isEmpty) {
-      logs
-    } else {
-      logs.filter(f => !deletedFiles.contains(f.path))
-    }
-  }
+  override def shouldRetain(log: SinkFileStatus): Boolean = true
```

Comment:
How about we still keep the `DELETE_ACTION` here?
```
if (log.action == FileStreamSinkLog.DELETE_ACTION) {
  false
} else {
  true
}
```
I also noticed that it's not implemented now, but it seems the only way to delete logs when compacting. I think it's also the original design goals. cc @zsxwing who is the author for SPARK-14678

Reply:
Please note that after the patch we are no longer able to match ADD_ACTION and DELETE_ACTION in compaction, so that's the responsibility of "reader" to exclude entities which have both ADD_ACTION and DELETE_ACTION, even we take DELETE_ACTION into account.

Returning `false` for DELETE_ACTION here simply drops the entities for DELETE_ACTION without dropping ADD_ACTION, hence some files would be resurrected on the reader side after compaction, making problems.

DELETE_ACTION is introduced years ago and we still haven't touched it. I think we should just defer and shouldn't deal with the thing we don't have and also don't have a clear plan to address. Once we design the new behavior, we would know what we should do. Dealing with something according to the imagination blocks us to move forward.

Reply:
Make sense. I also did some investigation and searched other thrid-party libraries, seems the DELETE_ACTION is not used. I think it's OK to design new behavior when we need it.